### ISSUE TITLE
tmux server exits with allow-rename enabled

### ISSUE THREAD

**Issue by c-14** [tmux/tmux#2188]
### Issue description

Since I upgraded tmux to version 3.1, the tmux server occasionally exits unexpectadly when allow-rename is on. From what I can see in the logs, the error seems to be
```
1588249595.658334 fatal: missing option automatic-rename
```
I can reproduce this issue on git master (and my logs are of git master). I tried running a git bisect to find out which commit caused the issue and narrowed down the commits to the following
```
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
bbe8ebf9c26e45fd8c402627b84b3646db445d45
f18cd5b19c3c8e3941c714f9d7e51914fb5d81f0
68d59a16cec91d2c2747c8c2862560e89f8f2d31
b598bbcc2e8b26855e4d34dfff9c222c28080cd7
eb57cbcc296b10d8d9ea41930ab402717c800f9c
We cannot bisect more!
```
which I cannot build on my system since make complains about a missing separator. Judging by the commit messages it's probably commit f18cd5b19c3c8e3941c714f9d7e51914fb5d81f0

### Required information

```
% uname -sp && tmux -V && echo $TERM
Linux AMD Ryzen 5 1600 Six-Core Processor
tmux 3.1
screen-256color
```
$TERM outside of tmux is rxvt-256color

Minimal config:
```
set -g allow-rename on
```

Server log:
https://gist.github.com/c-14/6c694adab88cc6576cdd7bdf5be208b0#file-tmux-server-15764-log

Client log:
https://gist.github.com/c-14/6c694adab88cc6576cdd7bdf5be208b0#file-tmux-client-15762-log

I've added the tmux logs as a gist because the logs are very long, but I can paste them here instead if you prefer.

**Comment by nicm** [comment]
Yes, try this please:
~~~~
--- a/input.c
+++ b/input.c
@@ -2354,7 +2354,7 @@ input_exit_rename(struct input_ctx *ictx)
                return;

        if (ictx->input_len == 0) {
-               oe = options_get(wp->window->options, "automatic-rename");
+               oe = options_get_only(wp->window->options, "automatic-rename");
                if (oe != NULL)
                        options_remove(oe);
                return;

**Comment by c-14** [comment]
That seems to fix the issue, thank you very much.

**Comment by nicm** [comment]
I will release 3.1b at some point to fix this but I will wait a while to see if there is anything else comes up.

**Comment by nbari** [comment]
latest tmux in macOS is crashing:

```
$ brew info tmux
tmux: stable 3.1a (bottled), HEAD
Terminal multiplexer
https://tmux.github.io/
/usr/local/Cellar/tmux/HEAD-e078f97 (9 files, 894KB) *
  Built from source on 2020-05-02 at 10:51:56
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/tmux.rb
```

Could it be related?

**Comment by nicm** [comment]
Do you have `allow-rename` turned on?

**Comment by nbari** [comment]
Hi @nicm, no but In my try to check for it by doing:

    tmux kill-server && tmux -vv new

And then
  
    tmux display-message -a

Tmux crashes and from the logs, last line I get this:

    fatal: tty_clamp_area: y too big, 4294967295 > 0

**Comment by nicm** [comment]
OK please open a new issue with full details and the full log.

**Comment by nicm** [comment]
I have released 3.1b now. Thanks!
